openedx/core/djangoapps/course_date_signals/handlers.py

```diff
@@ -91,13 +91,25 @@ def _get_custom_pacing_children(subsection, num_weeks):
     Return relative date items for the subsection and its children
     """
     items = [subsection]
+    all_problems_are_ora = True
+    has_empty_unit = True
     section_date_items = []
     while items:
         next_item = items.pop()
+        if next_item.category != 'sequential' and next_item.category != 'vertical':
+            has_empty_unit = False
```

Comment:
The `has_empty_unit` logic feels a bit indirect. You basically want to check that you have at least one ORA block, and then also if all of them are ORA. So you have two separate checks for that, one of which ("at least one ORA") is not labelled anything to do with ORA. (If I'm understanding the logic right.)

Is there a way to make that more explicit? Like keep a `has_ora` or `ora_count` variable, instead of `has_empty_unit`? Or throw every non-hierarchy-item's category into a `set` and then at the end just be like `if content_categories == {'openassessment'}:`.